gmso/tests/test_units.py

```diff
@@ -0,0 +1,223 @@
+        assert usys.convert_parameter(parameter, n_decimals=0) == "680"
+        parameter = typed_ethane.angles[0].angle_type.parameters["theta_eq"]
+        assert usys.convert_parameter(parameter, name="theta_eq") == "110.700"
+        parameter = typed_ethane.dihedrals[0].dihedral_type.parameters["c0"]
+        assert usys.convert_parameter(parameter) == "0.150"
+
+    def test_get_parameter_dimension(self):
+        from gmso.utils.units import get_parameter_dimension
+
+        assert (
+            get_parameter_dimension(1 * u.kJ / u.mol / u.nm, "(energy)")
+            == u.kJ / u.mol
+        )
+        assert get_parameter_dimension(1 * u.kJ / u.nm, "(length)") == u.nm
+        assert (
+            get_parameter_dimension(1 * u.kJ / u.mol / u.nm, "(length)") == u.nm
+        )
+
+    def test_convert_to_unit_system(self):
+        # TODO: discuss if we would want a function to convert a whole
+        # topology at once.
+        # convert a whole topology to a specific unit system
+        pass
+
+    def test_overwrite_unit_output(self):
+        # TODO: make a better usys.convert_parameter function to handle eq angles
+        # write out the parameters that have specific outputs
+        pass
+
+    def test_generate_unit_styles(self):
+        # TODO: write all unit systems for these engines.
+        # look at libary of unit styles for lammps, gromacs, hoomd, gomc
+        pass
+
+    def test_lj_units(self, typed_ethane):
+        # write out unit styles from ljUnitSystem and a dictonary of non-dimesnional values
+        usys = LAMMPS_UnitSystems("lj")
+        bond_parameter = typed_ethane.bonds[0].bond_type.parameters["k"]
+        errorStr = (
+            "Missing conversion_factorDict for a dimensionless unit system."
+        )
+        with pytest.raises(ValueError, match=errorStr):
+            usys.convert_parameter(bond_parameter, conversion_factorDict=None)
+        cfactorDict = {"energy": 0.276144 * u.kJ / u.mol, "length": 0.35 * u.nm}
+        errorStr = f"Missing dimensionless constant in conversion_factorDict {cfactorDict}"
+        with pytest.raises(ValueError, match=re.escape(errorStr)):
+            usys.convert_parameter(
+                bond_parameter, conversion_factorDict=cfactorDict
+            )
+        cfactorDict["charge"] = 1 * u.coulomb
+        cfactorDict["mass"] = 12.011 * u.amu
+        outStr = usys.convert_parameter(
+            bond_parameter, conversion_factorDict=cfactorDict
+        )
+        assert outStr == str(round(284512.0 / 0.276144 * 0.35**2, 3))
+        outStr = usys.convert_parameter(
+            typed_ethane.sites[0].atom_type.mass,
+            conversion_factorDict=cfactorDict,
+        )
+        assert outStr == f"{(round(1.000, 3)):.3f}"
+
+    def test_get_units(self, typed_ethane):
+        # get the unit system used for a topology
+        usys = LAMMPS_UnitSystems("real")
+        typed_ethane.system = usys
+        assert typed_ethane.system == usys
```

Comment:
```suggestion
        typed_ethane.unit_system = usys
        assert typed_ethane.unit_system == usys
```

Reply:
Done in [f21a6c2](https://github.com/mosdef-hub/gmso/pull/747/commits/f21a6c2b74d3bd172ccbc6d87ba3eec5773299a3)